### PR TITLE
feat: UI overhaul

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

A UI overhaul, including
- more accessible matches
- headers in the sidebar to separate and explain match types

They're based on [this Figma file](https://www.figma.com/file/5RM0QzmQe8vCcVY9cX4KLL/Accessibility?node-id=619%3A11119), with one conscious alteration – I've darkened the orange match colour to `300` to allow the icon colour to remain white in all cases.

## How to test

These changes mostly concern the UI – there are no new unit tests. See success criteria.

Here's how it looks downstream in Composer, the Guardian's CMS:
 
Screenshot 2022-07-25 at 10 44 40

To test yourself in Composer, follow the instructions introduced in [this PR.](https://github.com/guardian/prosemirror-typerighter/pull/213)

## How can we measure success?

Users find the new UI more accessible, the filter options are more clearly signposted, and it's clearer what each match colour means.

### CODE REVIEW

**Comment by SHession** [comment]
There is a slight wiggle on hover on "Review" decorations:

File: package.json [hunk 1]
```diff
@@ -82,6 +84,9 @@
     "whatwg-fetch": "^3.0.0"
   },
   "dependencies": {
+    "@emotion/react": "^11.4.1",
+    "@guardian/src-checkbox": "^3.0.1",
```
Comment (SHession) [inline package.json:88]: Are we using `src-checkbox` for anything?

**Review by SHession — COMMENTED** [review]
A few comments to consider about the appearance and a few CODE suggestions.

File: vite.config.ts [hunk 2]
```diff
@@ -3,27 +3,37 @@ import packageJson from "./package.json";
...
+  plugins: [
+    react({
+      jsxImportSource: "@emotion/react",
+      babel: {
+        plugins: ["@emotion/babel-plugin"]
+      }
+    })
+  ],
+  esbuild: {
+    logOverride: { "this-is-undefined-in-esm": "silent" }
```
Comment (jonathonherbert) [inline vite.config.ts:15]: Sadly necessary to ensure that our jsx transforms don't cause warnings in our output.

**Review by rebecca-thompson — APPROVED** [review]
Have tested this locally and it looks good to me

File: index.html [hunk 3]
```diff
@@ -7,7 +7,7 @@
   <meta http-equiv="X-UA-Compatible" content="ie=edge" />
   <title>Document</title>
   <script type="module" src="./pages/index.ts"></script>
-  <script src="https://unpkg.com/prosemirror-dev-tools@1.4.0/dist/umd/prosemirror-dev-tools.min.js"></script>
```
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply index.html]: No – thanks for spotting! Suspect I was on a high-latency connection and blocked by this script.

A good reminder to add this as a module and remove a networking dependency.

**Pull request merged**